### PR TITLE
vina homework 2

### PR DESCRIPTION
**Pull request by vinatsx**

### CODE REVIEW

File: Profiles/ProfilesAPI.py [hunk 1]
```diff
@@ -1,7 +1,58 @@
...
+@profiles_api.route('/<int:id>/', methods = ["GET"])
+def getid(id):
+    if id > len(db)-1:
+        return jsonify({"Status" : "Error", "Message" : "No such ID"})
+
+    else: 
+        det = db[id]
+        return jsonify({"Status" : "Success", "Details" : det})
+
+@profiles_api.route('/profiles', methods = ["POST"])
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:18]: This makes the endpoint /profiles/profiles due to the prefix on blue print. Maybe due to unclear instructions

File: Profiles/ProfilesAPI.py [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
+def postid():
+    newprofile = {}
+    newprofname = request.args.get('name')
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:21]: req.args return to url argument (?name=..) you should be looking for req.form for form body in a POST request

File: Profiles/ProfilesAPI.py [hunk 3]
```diff
@@ -1,7 +1,58 @@
...
+@profiles_api.route('/<int:id>/', methods = ["DELETE"])
+def deleteid(id):
+    if id > len(db)-1:
+        return jsonify({"Status" : "Error", "Message" : "No such ID"})
+
+    else:
+        deleteddetails = db[id]
+        del db[id]
+
+        return jsonify({"Status" : "Success", "New db" : db, "Deleted profile" : deleteddetails})
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:38]: Not a general practice to return the rest of db on delete

File: Auth/AuthAPI.py [hunk 4]
```diff
@@ -1,7 +1,59 @@
...
 sys.path.append("../")
 
 auth_api = Blueprint("auth", __name__)
+
+
+@auth_api.route('/register', methods = ["POST"])
+def reg():
+    
+    def regdetails(argsform):
+        username = argsform.get('username')
```
Comment (nobelang78) [inline Auth/AuthAPI.py:15]: Should be request.form, same as issue below

**Review by codecodecoder78 — COMMENTED** [review]
Some small errors, mainly between request.args and request.form. Otherwise is ok

**Pull request closed without merging**